lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -346,81 +372,65 @@ def main():
     changed = False
     if not os.path.exists(config_file):
         config_file = None
-    if db_exists(cursor, db):
-        if state == "absent":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
+
+    existence_list = []
+    non_existence_list = []
+    for each_database in db:
+        if db_exists(cursor, [each_database]):
+            existence_list.append(each_database)
+        else:
+            non_existence_list.append(each_database)
+
+    if state == "absent":
+        if module.check_mode:
+            module.exit_json(changed=bool(existence_list), db=db_name, db_list=db)
+        try:
+            changed = db_delete(cursor, existence_list)
+        except Exception as e:
+            module.fail_json(msg="error deleting database: %s" % to_native(e))
+        module.exit_json(changed=changed, db=db_name, db_list=db)
+    elif state == "present":
+        if module.check_mode:
+            module.exit_json(changed=bool(non_existence_list), db=db_name, db_list=db)
+        changed = False
+        if non_existence_list:
             try:
-                changed = db_delete(cursor, db)
+                changed = db_create(cursor, non_existence_list, encoding, collation)
             except Exception as e:
-                module.fail_json(msg="error deleting database: %s" % to_native(e))
-            module.exit_json(changed=changed, db=db_name)
-
-        elif state == "dump":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
-            rc, stdout, stderr = db_dump(module, login_host, login_user,
-                                         login_password, db, target, all_databases,
-                                         login_port, config_file, socket, ssl_cert, ssl_key,
-                                         ssl_ca, single_transaction, quick, ignore_tables)
-            if rc != 0:
-                module.fail_json(msg="%s" % stderr)
-            else:
-                module.exit_json(changed=True, db=db_name, msg=stdout)
-
-        elif state == "import":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
+                module.fail_json(msg="error creating database: %s" % to_native(e),
+                                 exception=traceback.format_exc())
+        module.exit_json(changed=changed, db=db_name, db_list=db)
+    elif state == "dump":
+        if non_existence_list and not all_databases:
+            module.fail_json(msg="Cannot dump database(s) %r - not found" % (', '.join(non_existence_list)))
+        if module.check_mode:
+            module.exit_json(changed=not bool(non_existence_list), db=db_name, db_list=db)
```

Comment:
```suggestion
            module.exit_json(changed=True, db=db_name, db_list=db)
```
So behavior is the same as without `check_mode`.

Reply:
fixed